pkg/network/protocols/http2/protocol.go

```diff
@@ -265,17 +266,30 @@ func (p *Protocol) PostStart(mgr *manager.Manager) error {
 	return p.dynamicTable.postStart(mgr, p.cfg)
 }
 
+func getMap(mgr *manager.Manager, name string) (*ebpf.Map, error) {
+	m, _, err := mgr.GetMap(name)
+	if err != nil {
+		return nil, fmt.Errorf("error getting %q map: %s", name, err)
+	}
+	if m == nil {
+		return nil, fmt.Errorf("%q map is nil", name)
+	}
+	return m, nil
+}
+
 func (p *Protocol) updateKernelTelemetry(mgr *manager.Manager) {
-	mp, _, err := mgr.GetMap(telemetryMap)
+	mp, err := getMap(mgr, telemetryMap)
 	if err != nil {
-		log.Warnf("unable to get http2 telemetry map: %s", err)
+		log.Warn(err)
 		return
 	}
 
-	if mp == nil {
-		log.Warn("http2 telemetry map is nil")
+	tlsMap, err := getMap(mgr, tlsTelemetryMap)
```

Comment:
shouldn't we perform it only if tls is enabled? I mean do we want that each time we will try to get the tls map as well to the plain map?

Reply:
this is a good question, the protocol does not know if tls is enabled, this is in our todo list.
For example today we're loading all probes and maps (including TLS probes and maps) if http2 is enabled, regardless of the state of any of the TLS capabilities.
I'd suggest to postpone the change you're suggesting to the PR that will make the protocols to be aware of the TLS state.

Reply:
sounds good as long we it is in our todo list :)